ginivision/src/main/java/net/gini/android/vision/internal/util/UriHelper.java

```diff
@@ -45,4 +47,28 @@
             }
         }
     }
+
+    /**
+     * Retrieves the filename of a Uri, if available.
+     *
+     * @param uri     a {@link Uri} pointing to a file
+     * @param context Android context
+     * @return the filename
+     * @throws IllegalStateException if the Uri is not pointing to a file or the filename was not
+     *                               available
+     */
+    @NonNull
+    public static String getFilenameFromUri(@NonNull final Uri uri, @NonNull final Context context) {
+        Cursor cursor = null;
+        cursor = context.getContentResolver().query(uri, null, null, null, null);
+        if (cursor == null) {
+            throw new IllegalStateException("Could not retrieve a Cursor for the Uri");
+        }
+        final int nameIndex = cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME);
```

Comment:
I guess this column always exists

Reply:
It doesn't hurt to be on the safe side with a library :)

Reply:
documentation of `getColumnIndexOrThrow` says:
>Returns the zero-based index for the given column name, or throws IllegalArgumentException if the column doesn't exist. If you're not sure if a column will exist or not use getColumnIndex(String) and check for -1, which is more efficient than catching the exceptions.

So to be on the save site maybe check with `getColumnIndex`

Reply:
Android... :)